web/packages/teleterm/src/services/appUpdater/appUpdater.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Teleport
+ * Copyright (C) 2025 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  autoUpdater,
+  AppUpdater as ElectronAppUpdater,
+} from 'electron-updater';
+import { ProviderRuntimeOptions } from 'electron-updater/out/providers/Provider';
+
+import Logger from 'teleterm/logger';
+
+import { ClientToolsUpdateProvider } from './clientToolsUpdateProvider';
+
+export class AppUpdater {
+  private readonly logger = new Logger('AppUpdater');
+
+  constructor() {
+    autoUpdater.setFeedURL({
+      provider: 'custom',
+      // Wraps ClientToolsUpdateProvider to allow passing getClientToolsVersion.
+      updateProvider: class extends ClientToolsUpdateProvider {
+        constructor(
+          options: unknown,
+          updater: ElectronAppUpdater,
+          runtimeOptions: ProviderRuntimeOptions
+        ) {
+          super(() => undefined, updater, runtimeOptions);
+        }
+      },
```

Comment:
How is `getClientToolsVersion` going to be used? On this branch alone this is hard to see as I assume it's implemented in one of the subsequent PRs.

But on its own it looks a bit cursed with the anonymous class. ;f The comment mentions allowing passing `getClientToolsVersion` but it's just not clear how this is related to what is happening below the comment.